app/use_cases/petition_plugin/detail_updater.rb

```diff
@@ -0,0 +1,14 @@
+module PetitionPlugin
+  class DetailUpdater
+    def perform(detail, attrs, version_attrs)
+      detail.update_attributes attrs 
+
+      current_version = detail.current_version
+      if current_version.nil? || current_version.body != version_attrs[:body]
+        detail.petition_detail_versions << PetitionPlugin::DetailVersion.new(version_attrs)
+      end
+
+      detail.save
```

Comment:
It's being *sent* because of the same reason I told you. It's a instance variable reference. And the view has access to the controller's instance variables.

Nevertheless, this is a Rails pattern and I don't think it is worth changing this right now.

But if you ask me, one option would be returning a object which contains a object response. the controller would expose `helper_methods` so the view would access them.

But again, there is no need to change it now. It is just a thought for future controllers / use cases.